evm/gas-service.js

```diff
@@ -0,0 +1,211 @@
+
+    const rpc = chain.rpc;
+    const provider = getDefaultProvider(rpc);
+
+    const wallet = await getWallet(privateKey, provider, options);
+    await printWalletInfo(wallet, options);
+
+    printInfo('Contract name', contractName);
+    printInfo('Contract address', GasServiceAddress);
+
+    const gasService = new Contract(GasServiceAddress, getContractJSON('IAxelarGasService').abi, wallet);
+
+    const gasOptions = await getGasOptions(chain, options, contractName);
+
+    printInfo('GasService Action', action);
+
+    if (prompt(`Proceed with action ${action} on chain ${chain.name}?`, yes)) {
+        return;
+    }
+
+    switch (action) {
+        case 'estimateGasFee': {
+            validateParameters({
+                isNonEmptyString: { destinationChain },
+                isValidAddress: { destinationAddress },
+                isNumber: { executionGasLimit },
+            });
+
+            const payload = options.payload || '0x';
+
+            const gasEstimate = await gasService.estimateGasFee(destinationChain, destinationAddress, payload, executionGasLimit);
+
+            printInfo('Gas Estimate', gasEstimate.toString());
+
+            break;
+        }
+
+        case 'updateGasInfo': {
+            validateParameters({
+                isNonEmptyStringArray: { chains },
+            });
+
+            let gasUpdates = await getGasUpdates(env, chain, chains);
+
+            const filteredChains = chains.filter((chain, i) => gasUpdates[i] !== null);
+            gasUpdates = gasUpdates.filter((update) => update !== null);
+
+            // Adding lowercase chain names for case insensitivity
+            filteredChains.forEach((destination, i) => {
+                if (destination.toLowerCase() !== destination) {
+                    filteredChains.push(destination.toLowerCase());
+                    gasUpdates.push(gasUpdates[i]);
+                }
+            });
+
+            const tx = await gasService.updateGasInfo(filteredChains, gasUpdates, gasOptions);
+
+            printInfo('Call updateGasInfo with following chains', filteredChains);
+            printInfo('TX', tx.hash);
+
+            const receipt = await tx.wait(chain.confirmations);
+
+            const eventEmitted = wasEventEmitted(receipt, gasService, 'GasInfoUpdated');
+
+            if (!eventEmitted) {
+                printWarn('Event not emitted in receipt.');
+            }
+
+            break;
+        }
+
+        default:
+            throw new Error(`Unknown action: ${action}`);
+    }
+}
+
+async function main(options) {
+    await mainProcessor(options, processCommand, false);
+}
+
+if (require.main === module) {
+    const program = new Command();
+
+    program.name('GasService').description('Script to manage GasService actions');
+
+    addBaseOptions(program, { address: true });
+
+    program.addOption(
+        new Option('-c, --contractName <contractName>', 'contract name').default('AxelarGasService').makeOptionMandatory(false),
```

Comment:
```suggestion
        new Option('-c, --contractName <contractName>', 'contract name').default('AxelarGasService'),
```
I think providing a default implies that it is not mandatory